server/src/output_stream/handler.rs

```diff
@@ -1,35 +1,54 @@
 use std::sync::Arc;
 
 use tokio::sync::mpsc::Sender;
-use tokio::sync::{Notify, RwLock};
```

Comment:
I think notify would've worked for you, you just need to split up when you call notified() and when you call await, as per https://docs.rs/tokio/1.7.0/tokio/sync/struct.Notify.html#method.notify_waiters

Reply:
Right, reading the documentation. Should have done that. 

New attempt here, this time only moving the creation around 52c2043